lib/wickedbase.pm

```diff
@@ -668,6 +671,24 @@ sub upload_wicked_logs {
     $self->upload_log_file("$dir_name.tar.gz");
 }
 
+=head2 do_barrier_create
+
+  do_barrier_create(<barrier_postfix> [, <test_name>] )
+
+Create a barier which can be later used to syncronize the wicked tests for SUT and REF.
+This function can be called statically. In this case the C<test_name> parameter is 
+mandatory.
+
+=cut
+sub do_barrier_create {
+    my ($self, $type, $test_name) = ref $_[0] ? @_ : (undef, @_);
```

Comment:
"/me shooting myself into the head while jumping from the cliff and keeping note - perl make me do so "

Reply:
because this function can be called `statically` like
```perl
wickedbase::do_barrier_create($type, $testname);
```
And as a member method:
```perl
my $wb = wickedbase->new();
$wb->do_barrier_create($type, $testname);
```
The difference is, that the later will get the first `$self` parameter, while the first one don't.

Reply:
and why we need to support both ways ?

Reply:
we have it here: 
 * https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/35980bf112836ad1799e68348235a410422e7601/tests/wicked/locks_init.pm#L21
where we use the static way and with this we need to pass the "testname".
But we also can use it within a test-case, where we don't need to pass the testname again:
 * https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/35980bf112836ad1799e68348235a410422e7601/tests/wicked/basic/ref/t12_DAD_dhcp4_detect.pm#L15